nim_status/lib/settings/types.nim

```diff
@@ -0,0 +1,214 @@
+    name* {.serializedFieldName("name").}: Option[string]
+    networks* {.serializedFieldName("networks/networks").}: JsonNode
+    # NotificationsEnabled indicates whether local notifications should be enabled (android only)
+    notificationsEnabled* {.serializedFieldName("notifications-enabled?").}: Option[bool]
+    photoPath* {.serializedFieldName("photo-path").}: string
+    pinnedMailservers* {.serializedFieldName("pinned-mailservers").}: Option[JsonNode]
+    preferredName* {.serializedFieldName("preferred-name").}: Option[string]
+    previewPrivacy* {.serializedFieldName("preview-privacy?").}: bool
+    publicKey* {.serializedFieldName("public-key").}: string
+    # PushNotificationsServerEnabled indicates whether we should be running a push notification server
+    pushNotificationsServerEnabled* {.serializedFieldName("push-notifications-server-enabled?").}: Option[bool]
+    # PushNotificationsFromContactsOnly indicates whether we should only receive push notifications from contacts
+    pushNotificationsFromContactsOnly* {.serializedFieldName("push-notifications-from-contacts-only?").}: Option[bool]
+    # PushNotificationsBlockMentions indicates whether we should receive notifications for mentions
+    pushNotificationsBlockMentions* {.serializedFieldName("push-notifications-block-mentions?").}: Option[bool]
+    rememberSyncingChoice* {.serializedFieldName("remember-syncing-choice?").}: Option[bool]
+    # RemotePushNotificationsEnabled indicates whether we should be using remote notifications (ios only for now)
+    remotePushNotificationsEnabled* {.serializedFieldName("remote-push-notifications-enabled?").}: Option[bool]
+    signingPhrase* {.serializedFieldName("signing-phrase").}: string
+    stickerPacksInstalled* {.serializedFieldName("stickers/packs-installed").}: Option[JsonNode]
+    stickersPacksPending* {.serializedFieldName("stickers/packs-pending").}: Option[JsonNode]
+    stickersRecentStickers* {.serializedFieldName("stickers/recent-stickers").}: Option[JsonNode]
+    syncingOnMobileNetwork* {.serializedFieldName("syncing-on-mobile-network?").}: Option[bool]
+    # SendPushNotifications indicates whether we should send push notifications for other clients
+    sendPushNotifications* {.serializedFieldName("send-push-notifications?").}: Option[bool]
+    appearance* {.serializedFieldName("appearance").}: uint
+    useMailservers* {.serializedFieldName("use-mailservers?").}: bool
+    usernames* {.serializedFieldName("usernames").}: Option[JsonNode]
+    walletRootAddress* {.serializedFieldName("wallet-root-address").}: Option[Address]
+    walletSetUpPassed* {.serializedFieldName("wallet-set-up-passed?").}: Option[bool]
+    walletVisibleTokens* {.serializedFieldName("wallet/visible-tokens").}: Option[JsonNode]
+    wakuBloomFilterMode* {.serializedFieldName("waku-bloom-filter-mode").}: Option[bool]
+    webViewAllowPermissionRequests* {.serializedFieldName("webview-allow-permission-requests?").}: Option[bool]
```

Comment:
It is my understanding that `{.serializedFieldName.}` is only needed if `json-serialization` is used, unless there is something I'm missing.

Reply:
Another way to prevent string mistakes would be use the stringified enum values here, like so:
```nim
chaosMode* {.serializedFieldName($SettingsEnum.ChaosMode).}: Option[bool]
```